sklearn/decomposition/tests/test_online_lda.py

```diff
@@ -0,0 +1,292 @@
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, learning_offset=30.,
+                                    random_state=rng)
+    for i in xrange(3):
+        lda.partial_fit(X)
+
+    correct_idx_grps = [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
+    for c in lda.components_:
+        top_idx = set(c.argsort()[-3:][::-1])
+        assert_true(tuple(sorted(top_idx)) in correct_idx_grps)
+
+
+def test_lda_dense_input():
+    """
+    Test LDA with dense input.
+    """
+    rng = np.random.RandomState(0)
+    X = rng.randint(5, size=(20, 10))
+    n_topics = 3
+    doc_topic_prior = 1. / n_topics
+    topic_word_prior = 1. / n_topics
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, random_state=rng)
+    X_trans = lda.fit_transform(X)
+    assert_true((X_trans > 0.0).any())
+
+
+def test_lda_fit_transform():
+    """
+    Test LDA fit_transform & transform
+    fit_transform and transform result should be the same
+    """
+    for method in ('online', 'batch'):
+        rng = np.random.RandomState(0)
+        n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+        lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                        topic_word_prior=topic_word_prior,
+                                        learning_method=method, random_state=rng)
+        X_fit = lda.fit_transform(X)
+        X_trans = lda.transform(X)
+        assert_array_almost_equal(X_fit, X_trans, 4)
+
+
+def test_lda_normalize_docs():
+    """
+    test sum of topic distribution equals to 1 for each doc
+    """
+    rng = np.random.RandomState(0)
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, normalize_doc=True,
+                                    random_state=rng)
+    X_fit = lda.fit_transform(X)
+    assert_array_almost_equal(X_fit.sum(axis=1), np.ones(X.shape[0]))
+
+
+@raises(ValueError)
+def test_lda_partial_fit_dim_mismatch():
+    """
+    test `n_features` mismatch in `partial_fit`
+    """
+    rng = np.random.RandomState(0)
+    n_topics = rng.randint(3, 6)
+    doc_topic_prior = 1. / n_topics
+    topic_word_prior = 1. / n_topics
+
+    n_col = rng.randint(6, 10)
+    X_1 = np.random.randint(4, size=(10, n_col))
+    X_2 = np.random.randint(4, size=(10, n_col + 1))
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, learning_offset=5.,
+                                    total_samples=20, random_state=rng)
+    for X in [X_1, X_2]:
+        lda.partial_fit(X)
```

Comment:
can you use ``assert_raises_regex`` on the second call to partial fit? That would be more explicit than the ``raises`` decorator.